site/static/api/device-registry-revised.yaml

```diff
@@ -0,0 +1,712 @@
+               $ref: '#/components/schemas/Extensions'
+            "adapters":
+               type: array
+               description: Only a single entry per type is allowed. If multiple entries for the same type are present it is handled as an error.
+               items:
+                  $ref: '#/components/schemas/Adapter'
+            "limits":
+               $ref: '#/components/schemas/ResourceLimit'
+            "trusted-ca":
+               $ref: '#/components/schemas/TrustedCA'
+
+      TrustedCA:
+         type: object
+         additionalProperties: false
+         required:
+            - "subject-dn"
+         properties:
+            "subject-dn":
+               type: string
+            "public-key":
+               type: string
+               format: byte
+            "cert":
+               type: string
+               format: byte
+
+      Adapter:
+         type: object
+         additionalProperties: true
+         required:
+            - type
+         properties:
+            "type":
+               type: string
+            "enabled":
+               type: boolean
+               default: false
+            "device-authentication-required":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+
+      ResourceLimit:
+         type: object
+         additionalProperties: false
+         properties:
+            "max-connections":
+               type: integer
+               default: -1
+
+# Devices schema
+
+      Device:
+         type: object
+         additionalProperties: false
+         properties:
+            "enabled":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+
+# Credentials
+
+      CredentialsSet:
+         type: object
+         additionalProperties:
+            $ref: '#/components/schemas/AuthenticationIdentity'
+
+      AuthenticationIdentity:
+         type: object
+         required:
+            - secrets
+         properties:
+            "enabled":
+               type: boolean
+               default: true
+            "secrets":
+               type: array
+               items:
+                  $ref: '#/components/schemas/TypedSecret'
+
+      TypedSecret:
+         oneOf:
+            - $ref: '#/components/schemas/PasswordSecret'
+            - $ref: '#/components/schemas/PSKSecret'
+            - $ref: '#/components/schemas/X509CertificateSecret'
+         discriminator:
+            propertyName: type
+            mapping:
+               "hashed-password": '#/components/schemas/PasswordSecret'
+               "psk": '#/components/schemas/PSKSecret'
+               "x509-cert": '#/components/schemas/X509CertificateSecret'
+
+      CommonSecret:
+         type: object
+         required:
+            - type
+         properties:
+            "type":
+               type: string
+            "not-before":
```

Comment:
the *auth-id* property is missing

Reply:
The `auth-id` is not part of the object itself, but the key of the map which holds this object.

Reply:
where is this defined? I cannot find it in the schema ...